sparse/densification.py

```diff
@@ -0,0 +1,140 @@
+import sys
+from numbers import Integral, Number
+from collections import Iterable
+
+import numpy as np
+
+from .utils import _get_broadcast_shape, SparseArray
+
+try:  # Windows compatibility
+    int = long
+except NameError:
+    pass
+
+try:
+    _max_size = sys.maxsize
+except NameError:
+    _max_size = sys.maxint
+
+
+class DensificationConfig(object):
+    def __init__(self, densify=None, max_size=None, min_density=None):
+        if isinstance(densify, (Iterable, DensificationConfig)):
+            self.make_copy_of(densify)
+            return
+
+        if max_size is None:
+            max_size = 10000
+
+        if min_density is None:
+            min_density = 0.25
+
+        if not isinstance(densify, bool) and densify is not None:
```

Comment:
Instead of allowing `densify` to be of mixed type (`None` and `bool`), you could [create a `Maybe` class that encapsulates the behaviour](https://stackoverflow.com/questions/9501148/booleans-have-two-possible-values-are-there-types-that-have-three-possible-valu)